plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterJSON.java

```diff
@@ -181,26 +164,73 @@ public void exportFooter(DBRProgressMonitor monitor) throws IOException
         out.write("\n");
     }
 
-    private void writeTextCell(@Nullable String value)
-    {
+    private void writeTextCell(@Nullable String value) {
         if (value != null) {
             getWriter().write("\"" + JSONUtils.escapeJsonString(value) + "\"");
         } else {
             getWriter().write("null");
         }
     }
 
-    private void writeCellValue(Reader reader) throws IOException
-    {
+    private void writeContentValue(
+        DBCSession session,
+        DBCResultSet resultSet,
+        DBDContent content
+    ) throws DBCException, IOException {
+        // Content
+        // Inline textual content and handle binaries in some special way
+
+        try {
+            DBDContentStorage cs = content.getContents(session.getProgressMonitor());
+            if (cs != null) {
+                if (ContentUtils.isTextContent(content)) {
+                    writeClob(content, cs);
+                } else {
+                    writeBlob(cs);
+                }
+            }
+        } finally {
+            DTUtils.closeContents(resultSet, content);
+        }
+    }
+
+    private void writeClob(
+        DBDContent content,
+        DBDContentStorage cs
+    ) throws IOException {
+        if (!exportJsonAsString && ContentUtils.isJsonContent(content)) {
+            try (Reader in = cs.getContentReader()) {
+                writeCellValue(in, false);
+            }
+        } else {
+            getWriter().write("\"");
+            try (Reader in = cs.getContentReader()) {
+                writeCellValue(in, true);
+            }
+            getWriter().write("\"");
+        }
+    }
+
+    private void writeBlob(DBDContentStorage cs) throws IOException {
+        getWriter().write("\"");
+        getSite().writeBinaryData(cs);
+        getWriter().write("\"");
+    }
+
+    private void writeCellValue(Reader reader, boolean escape) throws IOException {
         // Copy reader
-        char buffer[] = new char[2000];
-        for (;;) {
+        char[] buffer = new char[2000];
+        for (; ; ) {
```

Comment:
While we're at it, let's also fix this
```suggestion
        while (true) {
```